File: androidApp/core/domain/src/main/java/com/deadly/v2/core/domain/repository/ShowRepository.kt
package com.deadly.v2.core.domain.repository

import com.deadly.v2.core.model.Show
import com.deadly.v2.core.model.Recording
import kotlinx.coroutines.flow.Flow

/**
 * Domain repository interface for show operations.
 * 
 * Returns pure domain models (Show, Recording) and has no dependencies
 * on Android or data layer implementation details. This interface
 * represents the contract that data layer implementations must fulfill.
 */
interface ShowRepository {
    
    // Show queries
    suspend fun getShowById(showId: String): Show?
    suspend fun getShowsByIds(showIds: List<String>): List<Show>
    suspend fun getAllShows(): List<Show>
    fun getAllShowsFlow(): Flow<List<Show>>
    suspend fun getShowsByYear(year: Int): List<Show>
    suspend fun getShowsByYearMonth(yearMonth: String): List<Show>
    suspend fun getShowsByVenue(venueName: String): List<Show>
    suspend fun getShowsByCity(city: String): List<Show>
    suspend fun getShowsByState(state: String): List<Show>
    suspend fun getShowsBySong(songName: String): List<Show>
    suspend fun getTopRatedShows(limit: Int = 20): List<Show>
    suspend fun getRecentShows(limit: Int = 20): List<Show>
    suspend fun getShowsForDate(month: Int, day: Int): List<Show>
    suspend fun getShowCount(): Int
    
    // Navigation queries
    suspend fun getNextShowByDate(currentDate: String): Show?
    suspend fun getPreviousShowByDate(currentDate: String): Show?
    
    // Recording queries
    suspend fun getRecordingsForShow(showId: String): List<Recording>
    suspend fun getBestRecordingForShow(showId: String): Recording?
    suspend fun getRecordingById(identifier: String): Recording?
    suspend fun getTopRatedRecordings(minRating: Double = 2.0, minReviews: Int = 5, limit: Int = 50): List<Recording>
}

File: iosApp/deadly/Core/Domain/ShowRepository.swift
import Foundation

protocol ShowRepository: Sendable {
    // Single show
    func getShowById(_ showId: String) throws -> Show?
    func getShowsByIds(_ showIds: [String]) throws -> [Show]

    // Browse
    func getAllShows() throws -> [Show]
    func getShowCount() throws -> Int

    // Date queries
    func getShowsByYear(_ year: Int) throws -> [Show]
    func getShowsByYearMonth(_ yearMonth: String) throws -> [Show]
    func getShowsByDate(_ date: String) throws -> [Show]
    func getShowsForDate(month: Int, day: Int) throws -> [Show]  // "On This Day"

    // Location queries
    func getShowsByVenue(_ venueName: String) throws -> [Show]
    func getShowsByCity(_ city: String) throws -> [Show]
    func getShowsByState(_ state: String) throws -> [Show]

    // Featured
    func getTopRatedShows(limit: Int) throws -> [Show]

    // Chronological navigation
    func getNextShow(afterDate: String) throws -> Show?
    func getPreviousShow(beforeDate: String) throws -> Show?

    // Recording queries
    func getRecordingsForShow(_ showId: String) throws -> [Recording]
    func getBestRecordingForShow(_ showId: String) throws -> Recording?
    func getRecordingById(_ identifier: String) throws -> Recording?
    func getTopRatedRecordings(minRating: Double, minReviews: Int, limit: Int) throws -> [Recording]
}
